File: cases/taint/kotlin/return-relay-one-hop-negative/ReturnRelayOneHopNegative.kt
package dataflowbench

object ReturnRelayOneHopNegative {
    fun dfb_source(): Int { // DFB-SOURCE: return-one-hop-negative-input
        return 1
    }

    fun relay(value: Int): Int { // DFB-WITNESS: return-one-hop-negative-relay
        return value
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: return-one-hop-negative-sink

    fun run() {
        val result = relay(dfb_source())
        dfb_sink(0)
    }
}

File: cases/taint/go/return-relay-one-hop-negative/return_relay_one_hop_negative.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: return-one-hop-negative-input
	return 1
}

func relay(value int) int { // DFB-WITNESS: return-one-hop-negative-relay
	return value
}

func dfb_sink(value int) {} // DFB-SINK: return-one-hop-negative-sink

func run() {
	_ = relay(dfb_source())
	dfb_sink(0)
}
